pyproject.toml

```diff
@@ -13,6 +13,7 @@ readme = "./python-spec/README.md"
 dependencies = [
   "anndata",
   "attrs>=22.1",
+  "numba",
```

Comment:
`numba` doesn't work with Python 3.11 -- but, we're alreaady getting `numba` as an _indirect_ dependency so this is nothing new. (See also https://github.com/single-cell-data/TileDB-SOMA/issues/544)